src/pykiso/test_setup/dynamic_loader.py

```diff
@@ -353,6 +358,7 @@ def uninstall(self):
         """Deregister the import hooks, close all running threads, delete all instances."""
         log.internal_debug("closing and uninstalling all dynamic modules and loaders")
         self._stop_auxiliaries()
+        self._con_cache.delete_all_instances()
```

Comment:
You could also implement `delete_all_instances` within the con_cache's `__del__`
It would then be executed in the line below

Reply:
Same here, this never calls the custom __del__ method